### PR TITLE
Update release process documentation

### PR DESCRIPTION
**Pull request by ehashman**
```release-note
NONE
```

Updating the documentation on doing a release to better reflect the process I just went through.

/cc @mjudeikis @kwoodson

### CODE REVIEW

**Comment by ehashman** [comment]
/test upgrade-v5.2

flake #661

File: docs/release.md [hunk 1]
```diff
@@ -176,17 +176,27 @@ This step can be done together with step 2, but for first release you are doing
 git checkout -b osa.release.vx.testing
 ```
 
-Add vx.y target to `ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml`
+Add vx.y target to `ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml`
```
Comment (kwoodson) [inline docs/release.md]: Do we have anything that removes previous versions from CI once they are no longer valid?

**Comment by kwoodson** [comment]
Thanks @ehashman.  lgtm but will wait for @mjudeikis' comments

File: docs/release.md (hunk 1 above)
Comment (ehashman) [reply docs/release.md]: Yes, it's the bottom section of this doc ("cleaning an old release").

**Comment by ehashman** [comment]
Another flake #661 

/retest

**Pull request merged**